graphene/types/dynamic.py

```diff
@@ -11,7 +12,7 @@ class Dynamic(MountedType):
 
     def __init__(self, type, with_schema=False, _creation_counter=None):
         super(Dynamic, self).__init__(_creation_counter=_creation_counter)
-        assert inspect.isfunction(type)
+        assert inspect.isfunction(type) or isinstance(type, partial)
```

Comment:
I think a better check would be to use `callable` then it would catch both cases.

Reply:
You can't use callable because class types are callable.

Reply:
I could do ```not inspect.isclass(type) and callable(type)```

Reply:
Really though I was basing it off of https://github.com/graphql-python/graphene/blob/12d4dab7742c9408777380c390152d7cef0f838d/graphene/types/utils.py#L44

I wasn't sure if there were callable types we didn't want to support

Reply:
Ah yeah you're right. What you've done then looks good. Just needs some tests.